src/energy_comms/transform/msha.py

```diff
@@ -0,0 +1,41 @@
+"""Transform functions for the MSHA mine data."""
+from typing import Literal
+
+import geopandas
+import pandas as pd
+
+import energy_comms
+
+
+def transform(
+    raw_df: pd.DataFrame, census_geometry: Literal["state", "county", "tract"] = "tract"
+) -> pd.DataFrame:
+    """Standardize columns, filter for IRA coal mine criteria, join to census tract."""
+    df = raw_df.copy()
+    df.columns = df.columns.str.lower()
+    df["current_status_dt"] = pd.to_datetime(df["current_status_dt"].astype("string"))
+    # apply filters for IRA criteria
+    mask = (
+        (df.current_mine_status.isin(["Abandoned and Sealed", "Abandoned"]))
+        & (df.coal_metal_ind == "C")
+        & (df.current_status_dt.dt.year >= 2000)
+        & ~(df.longitude.isnull())
+        & ~(df.latitude.isnull())
+    )
+    df = df[mask]
+    # add geometry column to msha data
+    df = geopandas.GeoDataFrame(
+        df, geometry=geopandas.points_from_xy(df.longitude, df.latitude, crs=4269)
```

Comment:
The RFF report uses EPSG: 4269 and the Census data is from PUDL is 4269, so I went with that. But maybe there's another transformation that needs to happen to make these geometries planar and ready for overlap? There are about 150 records whose states don't match with the state indicated by it's corresponding Census record.